basic_module/account/src/traits/impls.rs

```diff
@@ -0,0 +1,150 @@
+use super::core::handles::*;
+use crate::account::{
+    add_balance, get_balance, get_previous_balance, get_previous_sequence, get_sequence, sub_balance,
+};
+use crate::{get_context, get_storage_id, Account, Action};
+use key::verify_ed25519;
+
+#[allow(dead_code)]
+pub struct Handler {}
+
+impl CheckTxHandler for Handler {
+    fn verify_transaction(tx: &Transaction) -> bool {
+        match &tx.action {
+            Action::Pay {
+                sender,
+                receiver: _,
+                quantity,
+            } => {
+                let sender_balance = get_balance(&sender);
+                let sender_sequence = get_sequence(&sender);
+                match (sender_balance, sender_sequence) {
+                    (Some(balance), Some(sequence)) => !(balance < tx.fee + quantity || sequence <= tx.seq),
+                    _ => false,
+                }
+            }
+            _ => false,
+        }
+    }
+}
+
+#[allow(dead_code)]
+pub struct Executer {}
+
+impl TransactionExecuter for Executer {
+    fn execute_transactions(context: &mut dyn SubStorageAccess, transactions: &[Transaction]) {
+        for tx in transactions {
+            match &tx.action {
+                Action::Pay {
+                    sender,
+                    receiver,
+                    quantity,
+                } => {
+                    let sender_balance = get_previous_balance(context, &sender).unwrap();
+                    let sender_sequence = get_previous_sequence(context, &sender).unwrap();
+                    if sender_balance >= tx.fee + quantity || sender_sequence > tx.seq {
+                        let _ = sub_balance(&sender, tx.fee + quantity);
```

Comment:
`sub_balance` checks if `sender_balance >= tx.fee + quantity` . Instead of double checking, let's utilize the result type.